File: python-client-sdk/src/sdk/models/shared/codat_public_api_models_platformcredentials_supportedfeaturestate.py
from __future__ import annotations
import dataclasses
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatPublicAPIModelsPlatformCredentialsSupportedFeatureState:
    feature_state: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('featureState'), 'exclude': lambda f: f is None }})
    feature_type: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('featureType'), 'exclude': lambda f: f is None }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsplatformcredentialssupportedfeaturestate.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Expose } from "class-transformer";


export class CodatPublicApiModelsPlatformCredentialsSupportedFeatureState extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "featureState" })
  featureState?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "featureType" })
  featureType?: string;
}